### PR TITLE
2 Middlewared Services refactored

### PR DESCRIPTION
**Pull request by sonicaj**
2 CRUD Services refactored for maximum consistency among the attributes they accept

### CODE REVIEW

File: src/middlewared/middlewared/plugins/pool.py [hunk 1]
```diff
@@ -725,15 +725,6 @@ class Config:
...
-        else:
-            data['dayweek'] = data['dayweek'].split(',')
-
-        if data['month'] == '*':
-            data['month'] = [str(value) for value in range(1, 13)]
-        else:
-            data['month'] = data['month'].split(',')
         return data
 
     async def validate_data(self, data, schema):
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/pool.py]: Forgot in first review: either add @private or make it a python private method `__validate_data`

File: src/middlewared/middlewared/plugins/pool.py [hunk 2]
```diff
@@ -767,44 +758,48 @@ class Config:
...
-            data['month'] = '*'
-        else:
-            data['month'] = ','.join(month)
+        weekdays = data.get('dayweek')
+        if weekdays and weekdays != '*':
+            try:
+                weekdays = [int(day) for day in weekdays.split(',')]
+            except ValueError:
+                verrors.add(
+                    f'{schema}.weekday',
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/pool.py]: Typo: dayweek

File: src/middlewared/middlewared/plugins/rsync.py [hunk 3]
```diff
@@ -354,15 +355,17 @@ class RsyncTaskService(CRUDService):
     class Config:
         datastore = 'tasks.rsync'
         datastore_prefix = 'rsync_'
+        datastore_extend = 'rsynctask.rsync_task_extend'
+
+    async def rsync_task_extend(self, data):
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/rsync.py]: @private

File: src/middlewared/middlewared/plugins/pool.py [hunk 4]
```diff
@@ -767,44 +758,48 @@ class Config:
...
-            verrors.add(
-                f'{schema}.month',
-                'This field is required'
-            )
-        elif len(month) == 12:
-            data['month'] = '*'
-        else:
-            data['month'] = ','.join(month)
+        weekdays = data.get('dayweek')
+        if weekdays and weekdays != '*':
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/pool.py]: Are we not limiting ourselves here? `*/4` is also a valid cron syntax.

File: src/middlewared/middlewared/plugins/pool.py [hunk 5]
```diff
@@ -767,44 +758,48 @@ class Config:
...
-            verrors.add(
-                f'{schema}.dayweek',
-                'This field is required'
-            )
-        elif len(dayweek) == 7:
-            data['dayweek'] = '*'
-        else:
-            data['dayweek'] = ','.join(dayweek)
+        months = data.get('month')
+        if months and months != '*':
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/pool.py]: Same here about limiting ourselves. Also, we seem to be duplicating quite a bit of work here it seems.

What do you think we make a reusable validator? Could that be worthwhile?

**Comment by william-gr** [comment]
@themylogin can you review as well, please?

File: src/middlewared/middlewared/plugins/pool.py [hunk 6]
```diff
@@ -767,48 +760,20 @@ class Config:
...
         Dict(
             'pool_scrub_create',
-            Int('pool', validators=[Range(min=1)]),
+            Int('pool', validators=[Range(min=1)], required=True),
             Int('threshold', validators=[Range(min=0)]),
             Str('description'),
-            List('dayweek', items=[Str('dayweek')]),
-            List('month', items=[Str('month')]),
-            Str('daymonth'),
+            Str('minute', validators=[CronField(place=0)]),
```
[the remark this answers is not in the document]
Comment (william-gr) [reply src/middlewared/middlewared/plugins/pool.py]: The schemas have some differences, or are we suggest we use Patch on top of that for all of them?

[the remark this answers is not in the document]
Comment (themylogin) [reply src/middlewared/middlewared/plugins/pool.py]: Are they? All cron schedules I've seen in FreeNAS include (and included only) minute, hour, day, month, weekday.

[the remark this answers is not in the document]
Comment (william-gr) [reply src/middlewared/middlewared/plugins/pool.py]: But part of the same schema we have other attributes, e.g. `disks`, `description` etc.

Or are you suggesting we have the shcedule within another attribute, e.g.
{
"description": "foo"
"schedule": {
  "minute": "*",
  "month": "*",
   ...
}
}

[the remark this answers is not in the document]
Comment (themylogin) [reply src/middlewared/middlewared/plugins/pool.py]: Yes, of course, another attribute that encapsulates a dict.

[the remark this answers is not in the document]
Comment (william-gr) [reply src/middlewared/middlewared/plugins/pool.py]: Sounds like a good idea to me. @sonicaj ?

**Pull request merged**